### PR TITLE
Facepile resize group

### PR DESCRIPTION
**Pull request by mattlind**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [X] Include a change request file using `$ npm run change`

#### Description of changes

Adding auto-sizing of the Facepile control to a given width or size of the containing element.

#### Focus areas to test

Test that Facepile looks good when dynamically resizing on on initial render using a given width or a discovered width.

### CODE REVIEW

**Comment by christiango** [comment]
Rather than bloating the existing component, does it make sense to create a ResizeableFacePile component instead? With this approach, it will bundle the resize group related code even for those that aren't using the resize behavior.

**Comment by micahgodbolt** [comment]
Interesting idea. We could treat facepile like a overflow set, and then wrap it in resize group for a responsive version.

**Comment by dzearing** [comment]
Agreed. Bundle size is a major concern, and facepile is pretty simple right now.

File: packages/office-ui-fabric-react/src/components/Facepile/Facepile.Props.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ export interface IFacepile {
 
 }
 
-export interface IFacepileProps extends React.Props<Facepile> {
+export interface IFacepileProps { // extends React.Props<Facepile> {
```
Comment (mattlind) [inline packages/office-ui-fabric-react/src/components/Facepile/Facepile.Props.ts]: remove comment

**Comment by mattlind** [comment]
Just got back to this.  I just saw @dzearing 's comment.  I called it FacepileResponsive intuitively but ResizableFacepile is fine with me if that's a preferred name.

File: packages/office-ui-fabric-react/src/components/Facepile/FacepileResponsive.tsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+      <div style={ { ...facepileWrapperProps } }>
+        <ResizeGroup
+          className={ className }
+          data={ facepileData }
+          onReduceData={ this._showLess }
+          onGrowData={ this._showMore }
+          onRenderData={ (data: IOverflowData) => {
+            return (
+              <Facepile {...this.props}
+                maxDisplayablePersonas={ data.primary.length }
```
Comment (mattlind) [inline packages/office-ui-fabric-react/src/components/Facepile/FacepileResponsive.tsx]: I don't love this.  Input would be nice.  Should Facepile take an overflow?

Comment (christiango) [reply packages/office-ui-fabric-react/src/components/Facepile/FacepileResponsive.tsx]: Shouldn't facepile just take in the exact faces to render in the top and the overflow? Or at least the top level count?

Comment (christiango) [reply packages/office-ui-fabric-react/src/components/Facepile/FacepileResponsive.tsx]: Probably worth aligning with the pattern established by command bar + overflow set. I can't recall what they did.

Comment (micahgodbolt) [reply packages/office-ui-fabric-react/src/components/Facepile/FacepileResponsive.tsx]: @christiango Matt and I just had a long discussion today about this....seems I totally missed this comment :D

I agree, I like the two array approach of main and overflow. We could consider deprecating maxDisplayedPersonas, and moving the the two array approach, making maxDisplayedPersonas and overflowPersonas mutually exclusive.

File: common/changes/@uifabric/example-app-base/FacepileResizeGroup_2017-10-03-18-34.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
```
Comment (christiango) [inline common/changes/@uifabric/example-app-base/FacepileResizeGroup_2017-10-03-18-34.json]: You didn't touch any package except for office-ui-fabric-react, there shouldn't be any changefiles for the other packages

**Comment by micahgodbolt** [comment]
I'm going to close this PR down for now. I know @mattlind was going to take a slightly different approach when he had time.

**Pull request closed without merging**